rush-plugins/rush-resolver-cache-plugin/src/afterInstallAsync.ts

```diff
@@ -0,0 +1,357 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { createHash } from 'node:crypto';
+import * as path from 'node:path';
+
+import type {
+  RushSession,
+  RushConfiguration,
+  RushConfigurationProject,
+  ILogger,
+  LookupByPath,
+  Subspace
+} from '@rushstack/rush-sdk';
+import type {
+  ISerializedResolveContext,
+  IResolverCacheFile
+} from '@rushstack/webpack-workspace-resolve-plugin';
+
+import { Async, FileSystem, PnpmShrinkwrapFile } from './externals';
+
+const MAX_LENGTH_WITHOUT_HASH: number = 120 - 26 - 1;
+const BASE32: string[] = 'abcdefghijklmnopqrstuvwxyz234567'.split('');
+
+// https://github.com/swansontec/rfc4648.js/blob/ead9c9b4b68e5d4a529f32925da02c02984e772c/src/codec.ts#L82-L118
+function createBase32Hash(input: string): string {
+  const data: Buffer = createHash('md5').update(input).digest();
```

Comment:
Hasn't md5 caused problems in later versions of Node?

Reply:
The issue was with MD4. This code needs to precisely replicate the behavior of PNPM with respect to calculating these names, because we are deriving the target of the existing installation.